torchaudio-contrib/io.py

```diff
@@ -0,0 +1,48 @@
+import torch
+import numpy as np
+import warnings
+from scipy.io import wavfile
+import librosa
+
+def load(filename, sr=44100, mono=True, norm=False, device='cpu', dtype=np.float32):
+    """
+    Generic wrapper for reading an audio file.
+    Different libraries offer different speeds for this, so this routine is the
+    'catch-all' for whatever read routine happens to work best
+
+    Tries a fast method via scipy first, reverts to slower librosa when necessary.
+    """
+    # first try to read via scipy, because it's fast
+    scipy_ok = False
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("error")    # scipy throws warnings which should be errors
+        try:
+            out_sr, signal = wavfile.read(filename)
+            scipy_ok = True
+        except wavfile.WavFileWarning:
+            print("read_audio_file: Warning raised by scipy. ",end="")
+
+    if scipy_ok:
+        if mono and (len(signal.shape) > 1):     # convert to mono (by truncating other channels)
+            signal = signal[:,0]
+
+        if isinstance(signal[0], np.int16):      # convert from ints to floats if necessary
```

Comment:
Firstly, this should be optional, and secondly, it's missing 8-bit and 24-bit conversion.

Reply:
- The 8-bit and 24-bit cases will trigger warnings from scipy and will kick the routine into librosa.